Application/src/Parser.java

```diff
@@ -0,0 +1,144 @@
+import model.sequenceDiagramParser.Processes;
+import model.sequenceDiagramParser.ContentArray;
+import javafx.scene.control.Alert;
+import com.google.gson.Gson;
+import model.sequenceDiagramParser.SequenceDiagramObject;
+
+
+/**
+ * @author Rashad Kamsheh & Isabelle Törnqvist
+ * @Version 1.0
+ * @since 2017-10-16
+ *
+ * made with usage of Gson
+ * Gson is released under the Apache 2.0 license.
+
+    Copyright 2008 Google Inc.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+ */
+
+public class Parser {
+
+    /** defining two different counter variables that will increment each time the diagram calling method
+    *that they are is called in order to differentiate between the imported diagrams
+    */
+    // odd number for the First Diagram counter
+    private static int counter = 1;
+    // even number for the parallel Diagram counter
+    private static int parallelCounter = 2;
+
+    /**
+     * static decelerations of String variables along with their proper formatting (to be returned for the backend)
+     */
+    private static String processes = "";
+    private static String properProcesses = "";
```

Comment:
why are they static and what are they used for

Reply:
+1 And methods below

Reply:
@timjon @Sebastian-Fransson  parseSequenceDiagram() needs to be static to be referenced from the main method.

Reply:
But the variables?